gui/src/components/Jobs/Form.js

```diff
@@ -22,56 +20,87 @@ const styles = theme => ({
   }
 })
 
-const JobFormLayout = ({
+const Form = ({
+  actionText,
+  dirty,
   isSubmitting,
   classes,
   handleChange,
   values,
+  touched,
+  errors,
   submitCount
 }) => {
   return (
-    <Fragment>
+    <React.Fragment>
       <Prompt
         when={values.json !== '' && submitCount === 0}
         message='You have not submitted the form, are you sure you want to leave?'
       />
-      <Form noValidate>
+      <formik.Form noValidate>
```

Comment:
Maybe during standup you could elaborate on the functionality of `formik`?

Reply:
@dimroc sure thing. It's something similar to simple form in Rails. It was already getting used on our existing forms to handle the submit event. I've just added some client side validation here to make sure the JSON is valid so that we can avoid passing around `shouldStringify`.